src/compiler/checker.ts

```diff
@@ -27141,8 +27158,13 @@ namespace ts {
             }
             const seen = createUnderscoreEscapedMap<Flags>();
 
-            for (const prop of node.properties) {
+            for (let index = 0; index < node.properties.length; ++index) {

```

Comment:
in my lint rule implementation I chose to iterate the properties array in reverse. That way you don't have `O(n)`